db/range_del_aggregator_v2.cc

```diff
@@ -214,53 +245,47 @@ void ReverseRangeDelIterator::Invalidate() {
   inactive_iters_.clear();
 }
 
-RangeDelAggregatorV2::RangeDelAggregatorV2(const InternalKeyComparator* icmp,
-                                           SequenceNumber /* upper_bound */)
-    : icmp_(icmp), forward_iter_(icmp, &iters_), reverse_iter_(icmp, &iters_) {}
-
-void RangeDelAggregatorV2::AddTombstones(
-    std::unique_ptr<FragmentedRangeTombstoneIterator> input_iter,
-    const InternalKey* smallest, const InternalKey* largest) {
-  if (input_iter == nullptr || input_iter->empty()) {
-    return;
-  }
-  if (wrapped_range_del_agg != nullptr) {
-    wrapped_range_del_agg->AddTombstones(std::move(input_iter), smallest,
-                                         largest);
-    // TODO: this eats the status of the wrapped call; may want to propagate it
-    return;
-  }
-  iters_.emplace_back(new TruncatedRangeDelIterator(std::move(input_iter),
-                                                    icmp_, smallest, largest));
-}
-
-bool RangeDelAggregatorV2::ShouldDelete(const ParsedInternalKey& parsed,
-                                        RangeDelPositioningMode mode) {
-  if (wrapped_range_del_agg != nullptr) {
-    return wrapped_range_del_agg->ShouldDelete(parsed, mode);
+bool RangeDelAggregatorV2::StripeRep::ShouldDelete(
```

Comment:
we need to make sure this doesn't impact the common case, i.e., no range tombstones. profiling the below command showed ~5% spent under this function

command (run a few times to populate the DB):
```
TEST_TMPDIR=/dev/shm perf record -g -e cycles ./db_bench -use_existing_db=true -benchmarks=seekrandomwhilewriting -seek_nexts=1024 -write_buffer_size=1048576 -target_file_size_base=1048576 -max_bytes_for_level_base=4194304 -compression_type=none -max_background_jobs=12 -threads=16 -benchmark_write_rate_limit=1048576 -duration=10
```

perf:
```
  Children      Self  Command          Shared Object        Symbol                                               ◆
...
-    5.33%     2.06%  db_bench         db_bench             [.] rocksdb::RangeDelAggregatorV2::StripeRep::ShouldD▒
   - 3.27% rocksdb::RangeDelAggregatorV2::StripeRep::ShouldDelete                                                ▒
        1.53% rocksdb::ReverseRangeDelIterator::Invalidate                                                       ▒
        1.46% rocksdb::ForwardRangeDelIterator::ShouldDelete                                                     ▒
   + 2.06% __clone                                                                                               ▒
```

Reply:
It looks like this was introduced before this PR. It might be worth fixing separately so we have something backport-able.

Reply:
Thanks for investigating this; I didn't realize that this would be so significant in the common case. Should be simple enough to shave off at least the 3% from child calls (not sure where the remaining 2% is coming from though). I'll also make a separate PR that can be backported onto the 5.18 branch.